src/commonMethods/commonFunctions.js

```diff
@@ -79,16 +173,318 @@ export const calcAvgFPS = function() {
   average = average.toFixed(2)
   this.$data.write('average', average)
 }
+
+/**
+ * This function is used to get screenshot
+ * @returns {Promise<void>}
+ */
+export const screenshot = async function() {
+  let url = `http://${constants.host}:80/Service/Snapshot/Capture?${moment().valueOf()}`
+  // create a new promise inside of the async function
+  let bufferData = new Promise((resolve, reject) => {
+    _http
+      .get(url, function(res) {
+        if (res.headers['content-length'] === undefined)
+          this.$log(
+            'Framework did not return a content-length! This will slow down the screenshot module.'
+          )
+        var buffers = []
+        var imageSize = res.headers['content-length']
+
+        res.on('data', function(chunk) {
+          buffers.push(Buffer.from(chunk))
+        })
+
+        res.on('end', function() {
+          return resolve(Buffer.concat(buffers, parseInt(imageSize)))
+        })
+      })
+      .on('error', function(e) {
+        return reject(e)
+      })
+  })
+
+  // wait for the promise to resolve
+  let result = await bufferData
+  this.$data.write('screenshotResult', result)
+}
+/**
+ * This function is used to kill the process
+ * @param process
+ */
+export const killProcess = function(process) {
```

Comment:
Aren't there JSONRPC-calls for doing things like this? Also, a method like this won't work in the browser version. cc @wouterlucas